lib/Server.js

```diff
@@ -986,17 +986,20 @@ class Server {
     }
   }
 
-  // send stats to a socket or multiple sockets
+  // Send stats to a socket or multiple sockets
   sendStats(sockets, stats, force) {
     const shouldEmit =
       !force &&
       stats &&
       (!stats.errors || stats.errors.length === 0) &&
+      (!stats.warnings || stats.warnings.length === 0) &&
```

Comment:
Here we fix bug, when if we have previously warning and run invalidate without change code (for example using `API` to invalidate), overlay with warnings hid and did not show again